ethereum-explorer/src/map_contract_events.rs

```diff
@@ -0,0 +1,56 @@
+mod pb;
+use crate::pb::eth::event::v1::Events;
+use crate::pb::eth::event::v1::Event;
+use substreams_ethereum::pb::eth::v2::Block;
+use substreams::Hex;
+use substreams_ethereum::pb::eth::v2::Log;
+use substreams_ethereum::pb::eth::v2::TransactionTrace;
+
+pub fn map_contract_events(contract_address: String, blk: &Block) -> Events {
+    let mut events: Vec<Event> = Vec::new();
+
+    for tr in &blk.transaction_traces {
```

Comment:
Failed transactions/failed calls are not handled properly, was this wanted?

Reply:
Fixed